ykrt/build.rs

```diff
@@ -1,13 +1,16 @@
 pub fn main() {
     ykbuild::apply_llvm_ld_library_path();
-
+    println!("cargo:rerun-if-env-changed=YKB_TRACER");
     // Always compile in the LLVM JIT compiler.
     println!("cargo:rustc-cfg=jitc_llvm");
     // Always compile in our bespoke JIT compiler.
     println!("cargo:rustc-cfg=jitc_yk");
-    // Always compile in the HWT tracer.
-    println!("cargo:rustc-cfg=tracer_hwt");
     // FIXME: This is a temporary hack because LLVM has problems if the main thread exits before
     // compilation threads have finished.
     println!("cargo:rustc-cfg=yk_llvm_sync_hack");
+    if std::env::var("YKB_TRACER") == Ok("sw".to_owned()) {
```

Comment:
I think this should be something like:

```rust
match env::var("YKB_TRACER").as_ref() {
  Ok("swt") => ...,
  Ok("hwt") | Err(env::VarError::NotPresent) => ...
  Ok(x) => panic!("Unknown tracer {x}"),
  Err(_) => panic!("Invalid value for YKB_TRACER")
}
```

so that invalid values of `YKB_TRACER` aren't silently (confusingly) ignored.

At some point the default should probably move from hwt -> swt. But not yet I suppose!

Reply:
updated 👉 54e80f246c153ae1f1ae91e0150b4c24cb7de16d